05/DB.test.js

```diff
@@ -1 +1,151 @@
-import DB from './DB';
\ No newline at end of file
+import DB from "./DB";
+
+describe("class DB", () => {
+  it(`Creates new instance of the class | WHEN: "new" operator is used.`, () => {
+    const db = new DB();
+
+    function createNewDB(db) {
+      if (db) {
+        return true;
+      }
+
+      return false;
+    }
+
+    expect(createNewDB(db)).toBe(true);
+  });
+
+  describe("insert(data)", () => {
+    // scenario
+    it("REJECTS: the promise | WHEN: data ID is not a number", async () => {
+      // given
+      const db = new DB();
+      // Czy nie brakuje tutaj when?
+      // then
+      await expect(
+        db.insert({ id: "not a number", a: 1, b: 2 })
+      ).rejects.toBeTruthy();
+    });
+
+    it("REJECTS: the promise | WHEN: data ID is a duplicate", async () => {
+      const db = new DB();
+
+      await db.insert({ id: 5, a: 1, b: 2 });
+      await expect(db.insert({ id: 5, a: 2, b: 3 })).rejects.toBeTruthy();
+    });
+
+    it("REJECTS: the promise | WHEN: there's no data", async () => {
+      const db = new DB();
+
+      await expect(db.insert()).rejects.toBeTruthy();
+    });
+
+    it("Grants new ID | WHEN: data don't have an ID", async () => {
+      const db = new DB();
+
+      await db.insert({ a: 1, b: 2 });
+      expect(db._rows[0].id).toBeTruthy();
+    });
+
+    it("PUSHES: data to the array", async () => {
+      const db = new DB();
+      await db.insert({ a: 1, b: 2 });
+      expect(db._rows.length > 0).toBeTruthy();
```

Comment:
Trochę nadużywasz `toBeTruthy()` ;P
Czy dodając jeden element do DB faktycznie długość ma byc większa od 0?
Czy jeśli będzie to 10 lub 2 to czy faktycznie całość działa prawidłowo?
NIE, bo dodaliśmy jeden element, wiec oczekujemy, że ilość elementów to 1 ;)